recirq/qml_lfe/circuit_blocks.py

```diff
@@ -0,0 +1,162 @@
+    """
+    mapped_circuit = _load_circuit("tsym_permute.txt").transform_qubits(
+        {cirq.GridQubit(0, 0): qubits[0], cirq.GridQubit(0, 1): qubits[1]}
+    )
+    rots = [cirq.Y(qubits[0]) ** params[0], cirq.Y(qubits[1]) ** params[1]]
+    return rots + list(mapped_circuit.all_operations())
+
+
+def _get_op(code):
+    if code <= 4 / 3:
+        return cirq.X ** 0.5
+    elif code <= 2 * (4 / 3):
+        return cirq.Y ** 0.5
+    elif code <= 3 * (4 / 3):
+        return cirq.PhasedXZGate(axis_phase_exponent=0.25, x_exponent=0.5, z_exponent=0)
+
+    raise ValueError("panic!")
+
+
+def scrambling_block(qubits, params):
+    """Create a scrambling block.
+
+    Randomly applies one of sqrt(X), sqrt(Y) and sqrt(W) on qubits followed
+    by a SycamoreGate.
+
+    Args:
+        qubits: The pair of qubits to create a scrambling block on.
+        params: List of two random numbers between 0 and 4 to decide on
+            single qubit operations for each qubit.
+
+    Returns:
+        A list of `cirq.Operations` realizing a scrambling block.
+    """
+    a_op = _get_op(params[0])
+    b_op = _get_op(params[1])
+    return [a_op(qubits[0]), b_op(qubits[1]), cirq_google.SycamoreGate()(*qubits)]
+
+
+def block_1d_circuit(qubits, depth, block_fn, random_source):
+    """Create a 1D block structure circuit using block_fn.
+
+    Alternate couplings layer by layer constructing a circuit that
+    looks like:
+
+    a --- b XXX c --- d XXX e --- f
+    a XXX b --- c XXX d --- e XXX f
+    a --- b XXX c --- d XXX e --- f
+    a XXX b --- c XXX d --- e XXX f
+
+
+    Where each XXX is constructed by calling `block_fn` on each pair of qubits,
+    using the next unused entry in `random_source`.
+
+    Args:
+        qubits: List of adjacent qubits.
+        depth: Number of layers to include in circuit.
+        block_fn: Python function that accepts qubit pairs and
+            parameters to generate interactions on those qubit pairs.
+        random_source: Python list or np.ndarray with shape:
+            [depth * len(qubits) // 2, n_params_for_block_fn].
+
+    Returns:
+        A 1d block circuit.
+    """
+    ops = []
+    i_random = 0
+    for t in range(depth):
+        for i in range(0 if t % 2 == 0 else 1, len(qubits) - 1, 2):
+            ops += block_fn([qubits[i], qubits[i + 1]], random_source[i_random])
+            i_random += 1
+
+    return cirq.Circuit(ops)
+
+
+def bell_pair_block(qubits):
+    """Creates a bell pair on qubits.
+
+    Enacts H(a) + CNOT(a, b) using SycamoreGates and single qubit operations.
+
+    Args:
+        qubits: The qubits to prepare the bell pair on.
+
+    Returns:
+         A list of `cirq.Operations` realizing a bell pair.
+    """
+    mapped_circuit = _load_circuit("syc_bell_pair.txt").transform_qubits(
+        {cirq.GridQubit(0, 0): qubits[0], cirq.GridQubit(0, 1): qubits[1]}
+    )
+    return mapped_circuit.all_operations()
+
+
+def un_bell_pair_block(qubits):
+    """Un-bell-pair on qubits.
```

Comment:
I would improve this statement.  Maybe "Uncomputes a bell pair on two qubits."?

Reply:
Done.